### PR TITLE
Export SOCKET_KEY from feathersjs/socket.io

### PR DESCRIPTION
**Pull request by deskoh**
Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] Add or edit tests to reflect the change. (Run with `npm test`.)
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/feathersjs-ecosystem/feathers-typescript/issues/29
https://github.com/feathersjs/feathers/blob/buzzard/packages/socketio/lib/index.js

The definition follows the typings in upcoming Feathers Crow:
https://github.com/feathersjs/feathers/blob/master/packages/socketio/index.d.ts

- [X] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

The `SOCKET_KEY` allows mocking of connection to test channel publication. e.g.

```ts
import { SOCKET_KEY } from '@feathersjs/socketio';

export default mockConnection(assertCb) {
  const conn = { };
  conn[SOCKET_KEY] = {
    emit: (eventName, result) => {
      if (assertCb) {
        assertCb(eventName, result);
      }
    }
  }
}
```

### CODE REVIEW

File: types/feathersjs__socketio/index.d.ts [hunk 1]
```diff
@@ -1,12 +1,22 @@
...
 // Definitions by: Jan Lohage <https://github.com/j2L4e>
+//                 Desmond Koh <https://github.com/deskoh>
 // Definitions: https://github.com/feathersjs-ecosystem/feathers-typescript
 // TypeScript Version: 2.8
 
-/// <reference types="feathersjs__socket-commons"/>
-import SocketIO = require("socket.io");
+/// <reference types="node" />
 
-export default function feathersSocketIO(callback?: (io: SocketIO.Server) => void): () => void;
```
Comment (uniqueiniquity) [inline types/feathersjs__socketio/index.d.ts]: Why did you remove these `export default` overloads in favor of an interface with an explicit `default` property? You should just be able to add `export const SOCKET_KEY: unique symbol;` and get the same effect.

File: types/feathersjs__socketio/index.d.ts [hunk 2]
```diff
@@ -1,12 +1,22 @@
...
-export default function feathersSocketIO(options: number | SocketIO.ServerOptions, callback?: (io: SocketIO.Server) => void): () => void;
-export default function feathersSocketIO(port: number, options?: SocketIO.ServerOptions, callback?: (io: SocketIO.Server) => void): () => void;
+import * as http from 'http';
```
Comment (uniqueiniquity) [inline types/feathersjs__socketio/index.d.ts]: Is there a reason you're importing this? It doesn't look like it's used anywhere.

**Review by uniqueiniquity — CHANGES_REQUESTED** [review]

File: types/feathersjs__socketio/index.d.ts (hunk 1 above)
Comment (deskoh) [reply types/feathersjs__socketio/index.d.ts]: See discussion https://github.com/feathersjs-ecosystem/feathers-typescript/issues/29
and related PR https://github.com/DefinitelyTyped/DefinitelyTyped/pull/34515.

File: types/feathersjs__socketio/index.d.ts (hunk 2 above)
Comment (deskoh) [reply types/feathersjs__socketio/index.d.ts]: I can't recall the issue it caused previously. But removing it together with `node` reference seemed fine now.

**Review by uniqueiniquity — APPROVED** [review]
